core/config.py

```
@@ -159,12 +159,35 @@ class StrategiesConfig(BaseModel):
+
+    @validator("cors_origins")
+    def validate_cors_origins(cls, v, values):
+        """Validate CORS origins - prevent wildcard with credentials."""
+        # If credentials are allowed (default True), wildcard is forbidden
+        allow_credentials = values.get("cors_allow_credentials", True)
+        if allow_credentials and "*" in v:
+            raise ValueError(
+                "CORS origins cannot contain '*' when credentials are allowed. "
+                "Use explicit origins like 'http://localhost:3000' or set "
+                "cors_allow_credentials=False (not recommended for authenticated APIs)."
+            )
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)